### PR TITLE
compiler: support RawSyscall.  os: implement Pipe for darwin.

### PR DESCRIPTION
**Pull request by dankegel**
At bare minimum, one should be able to write a short message to the write end of a Pipe, then read it from the read end.

I didn't see an obvious test like that upstream, so I threw one together.

### CODE REVIEW

**Comment by dkegel-fastly** [comment]
assert-test-linux looks like a false pass; if you look at the "test stdlib packages" step, there is an error.

It feels like this test found about three problems...?

**Comment by aykevl** [comment]
Huh, weird. Without further investigation, I'd look into #2412 which changes the `tinygo test` logic.

**Comment by dkegel-fastly** [comment]
And... maybe underneath pipe is all still a stub.

**Comment by dkegel-fastly** [comment]
I have a draft of unstubbed pipe on my local tree for mac passing this test, will finish when I get a chance.

**Comment by dkegel-fastly** [comment]
All is well except on linux, which is suffering from #2466 and #2474.
Once #2466 is fixed, this will go red; once #2474 is fixed, this will go green.

**Comment by dkegel-fastly** [comment]
Includes Ayke's suggested fix for RawSyscall now.

**Comment by deadprogram** [comment]
Please correct merge conflict @dankegel thank you.

File: compiler/compiler.go [hunk 1]
```diff
@@ -1435,7 +1435,8 @@ func (b *builder) createFunctionCall(instr *ssa.CallCommon) (llvm.Value, error)
 			return b.emitSV64Call(instr.Args)
 		case strings.HasPrefix(name, "(device/riscv.CSR)."):
 			return b.emitCSROperation(instr)
-		case strings.HasPrefix(name, "syscall.Syscall"):
+		case strings.HasPrefix(name, "syscall.Syscall") || strings.HasPrefix(name, "syscall.RawSyscall"):
+			// TODO: implement b.createRawSyscall for real to reduce overhead slightly
```
Comment (aykevl) [inline compiler/compiler.go]: Which overhead? TinyGo just calls syscalls directly without any stack switching overhead like in upstream Go (in fact, the call is inlined so there's no real way to optimize it further). So this comment seems misleading.

Also, note that `b.createRawSyscall` is a different 'raw' than `syscall.RawSyscall`.

**Review by aykevl — COMMENTED** [review]
Looks good to me, except for one nit.

**Comment by deadprogram** [comment]
Also, please resolve merge conflict @dankegel thank you!

File: compiler/compiler.go (hunk 1 above)
Comment (dankegel) [reply compiler/compiler.go]: Removed comment.

**Comment by deadprogram** [comment]
Thanks @dankegel now merging.

**Pull request merged**